process/smartContract/scDataGetter.go

```diff
@@ -38,15 +46,24 @@ func (scdg *scDataGetter) Get(scAddress []byte, funcName string, args ...[]byte)
 	if scAddress == nil {
 		return nil, process.ErrNilScAddress
 	}
+	if len(scAddress) < scdg.addrConverter.AddressLen() {
+		return nil, process.ErrInvalidRcvAddr
+	}
 	if len(funcName) == 0 {
 		return nil, process.ErrEmptyFunctionName
 	}
 
 	scdg.mutRunSc.Lock()
 	defer scdg.mutRunSc.Unlock()
 
+	vmType := scAddress[core.NumInitCharactersForScAddress-core.VMTypeLen : core.NumInitCharactersForScAddress]
```

Comment:
Don't we have a func for this? Each time I recheck if the slice is done ok.

Reply:
Done. moved to core.